internal/services/policy/remediation_resource_group.go

```diff
@@ -172,77 +176,80 @@ func resourceArmResourceGroupPolicyRemediationRead(d *pluginsdk.ResourceData, me
+		d.Set("policy_definition_id", props.PolicyDefinitionReferenceId)
+		d.Set("resource_discovery_mode", utils.NormalizeNilableString((*string)(props.ResourceDiscoveryMode)))
+
 	}
 
 	return nil
 }
 
 func resourceArmResourceGroupPolicyRemediationDelete(d *pluginsdk.ResourceData, meta interface{}) error {
-	client := meta.(*clients.Client).Policy.RemediationsClient
+	client := meta.(*clients.Client).Policy.PolicyInsightsClient
 	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.ResourceGroupPolicyRemediationID(d.Id())
+	id, err := policyinsights.ParseProviderRemediationID(d.Id())
 	if err != nil {
 		return err
 	}
 
 	// we have to cancel the remediation first before deleting it when the resource_discovery_mode is set to ReEvaluateCompliance
 	// therefore we first retrieve the remediation to see if the resource_discovery_mode is switched to ReEvaluateCompliance
-	existing, err := client.GetAtResourceGroup(ctx, id.SubscriptionId, id.ResourceGroup, id.RemediationName)
+	existing, err := client.RemediationsGetAtResourceGroup(ctx, *id)
 	if err != nil {
-		if utils.ResponseWasNotFound(existing.Response) {
+		if response.WasNotFound(existing.HttpResponse) {
 			return nil
 		}
 		return fmt.Errorf("retrieving %s: %+v", id, err)
 	}
 
-	if existing.RemediationProperties != nil && existing.RemediationProperties.ResourceDiscoveryMode == policyinsights.ReEvaluateCompliance {
-		// Remediation can only be canceld when it is in "Evaluating" status, otherwise, API might raise error (e.g. canceling a "Completed" remediation returns 400).
-		if existing.RemediationProperties.ProvisioningState != nil && *existing.RemediationProperties.ProvisioningState == "Evaluating" {
-			log.Printf("[DEBUG] cancelling the remediation first before deleting it when `resource_discovery_mode` is set to `ReEvaluateCompliance`")
-			if _, err := client.CancelAtResourceGroup(ctx, id.SubscriptionId, id.ResourceGroup, id.RemediationName); err != nil {
-				return fmt.Errorf("cancelling %s: %+v", id.ID(), err)
-			}
-
-			log.Printf("[DEBUG] waiting for the %s to be canceled", id.ID())
-			stateConf := &pluginsdk.StateChangeConf{
-				Pending: []string{"Cancelling"},
-				Target: []string{
-					"Succeeded", "Canceled", "Failed",
-				},
-				Refresh:    resourceGroupPolicyRemediationCancellationRefreshFunc(ctx, client, *id),
-				MinTimeout: 10 * time.Second,
-				Timeout:    d.Timeout(pluginsdk.TimeoutDelete),
-			}
-
-			if _, err := stateConf.WaitForStateContext(ctx); err != nil {
-				return fmt.Errorf("waiting for %s to be canceled: %+v", id.ID(), err)
+	if prop := existing.Model.Properties; prop != nil {
+		if mode := existing.Model.Properties.ResourceDiscoveryMode; mode != nil && *mode == policyinsights.ResourceDiscoveryModeReEvaluateCompliance {
+			// Remediation can only be canceld when it is in "Evaluating" status, otherwise, API might raise error (e.g. canceling a "Completed" remediation returns 400).
```

Comment:
```suggestion
			// Remediation can only be canceld when it is in "Evaluating" or "Accepted" status, otherwise, API might raise error (e.g. canceling a "Completed" remediation returns 400).
```